src/librustc_mir/interpret/machine.rs

```diff
@@ -15,6 +15,22 @@ use super::{
     Frame, Operand,
 };
 
+/// Data returned by Machine::stack_pop,
+/// to provide further control over the popping of the stack frame
+#[derive(Eq, PartialEq, Debug)]
+pub enum StackPopInfo {
+    /// Indicates that we have just started unwinding
+    /// as the result of panic
+    StartUnwinding,
+
+    /// Indicates that we performed a normal return
+    Normal,
```

Comment:
Why is there no equivalent of "resume" in here? I think it is because `Normal` is also used for that, but then the doc comment here is misleading.

I am still trying to understand why we need this enum, but I suppose for that I need to look at the Miri part of this PR. Anyway this seems like something that, if we can avoid it, we can clean up later.

Reply:
Resuming happens when we hit a `Resume` terminator - the `Machine` doesn't need to be involved in that.

The purpose of this enum is to allow Miri to signal that we've started unwinding (when we start panicking), and that we've completely stopped unwinding (when we reach a `catch` frame). The base interpreter has no knowledge of these, so it needs Miri to determine which frames are 'special' in that regard.

Reply:
But we also call the machine hook when we are unwinding, don't we? After all, otherwise it couldn't *stop* unwinding.

So `Normal` here means "do what you'd usually do -- return or unwind". It doesn't always mean return.

Reply:
@Aaron1011 this comment is not resolved yet.

Reply:
I'll update the doc comment